src/dependency_downloader.rs

```diff
@@ -287,6 +285,49 @@ fn transform_git_to_http(url: &str) -> String {
     }
 }
 
+fn check_recursiveness(dependency: &Dependency) -> Result<()> {
+    let dep_name =
+        sanitize_dependency_name(&format!("{}-{}", dependency.name(), dependency.version()));
+
+    let dep_dir = DEPENDENCY_DIR.join(dep_name);
+    if !dep_dir.exists() {
+        return Err(DownloadError::RecursivenessError(
+            "Dependency directory does not exists".to_string(),
+        ));
+    }
+
+    let mut git = Command::new("git");
+
+    let result = git
+        .args(["submodule", "update", "--init", "--recursive"])
+        .env("GIT_TERMINAL_PROMPT", "0")
+        .current_dir(&dep_dir)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped());
+
+    let status = result.status().unwrap();
+
+    if !status.success() {
+        println!("{}", "Dependency has no submodule dependency.".yellow());
+    }
+
+    let mut soldeer = Command::new("forge");
+
+    let result = soldeer
+        .args(["soldeer", "install"])
+        .current_dir(&dep_dir)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped());
+
+    let status = result.status().unwrap();
```

Comment:
Avoid `unwrap`, better to report the error instead of panicking? If don't want to return an error, then at least use `expect()` with a descriptive message.